apps/schematic/api/schematic_api/controllers/schema_controller_impl.py

```diff
@@ -62,3 +64,75 @@ def list_component_attributes(
     status = 200
 
     return result, status
+
+
+def get_node_dependencies(
+    node_label: str,
+    schema_url: str,
+    return_display_names: bool = True,
+    return_ordered_by_schema: bool = True,
+) -> list[str]:
+    """Gets the components that the input component is dependent on
+
+    Args:
+        node_label (str): The label of the component to get dependencies for
+        schema_url (str): The URL of the schema in json form
+        return_display_names (bool): Whether or not to return the display names of the component,
+          otherwise the label
+        return_ordered_by_schema (bool):Whether or not to order the components by their order in
+          the schema, otherwise random
+
+    Returns:
+        list[str]: A list of component labels or display names
+    """
+    schema_generator = SchemaGenerator(path_to_json_ld=schema_url)
+    return schema_generator.get_node_dependencies(
+        source_node=node_label,
+        display_names=return_display_names,
+        schema_ordered=return_ordered_by_schema,
+    )
+
+
+@handle_exceptions
+def list_node_dependencies(
+    node_label: str,
+    schema_url: str,
+    return_display_names: bool = True,
+    return_ordered_by_schema: bool = True,
+) -> tuple[Union[NodesPage, BasicError], int]:
+    """Lists the components that the input component is dependent on
```

Comment:
this should read as "Lists the attributes that the input attribute is dependent on"